woocommerce/woocommerce
# E2e/playwright tests en es ar

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:
Update `e2e` tests to allow them to be executed against an English and non-English language site





Closes https://github.com/woocommerce/woocommerce-quality/issues/529



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Pull down this branch
2. Create the environment as usual with `pnpm env:test`
3. Execute the e2e tests to ensure they are all operational with `pnpm test:e2e-pw`
4. Now, switch the language from English to Espanol by performing steps 5-14 below:
5. Navigate to `http://localhost:8086/wp-admin/`
6. Login with the usual admin username and password
7. Click on `Settings` then `General`
<img width="359" alt="image" src="https://github.com/woocommerce/woocommerce/assets/105309450/0706eea4-0a41-4f81-85fb-81544e7eae4a">

8. Select `Site Language` to be Espanol
9. Click `Save Changes` button
10. Check to see if the incomplete task (to update translations) has appeared:
<img width="715" alt="image" src="https://github.com/woocommerce/woocommerce/assets/105309450/28c2e604-6bb2-44ab-b0c4-619a7c8f5cf4">

11. If it hasn't then change the language back to English, Save changes (the button will now be in Espanol `Guardar cambios`) and the incomplete task should appear. Change the language back to Espanol
13. Click on this incomplete task button to navigate to the page to update translations
14. Click on the bottom button to update the translations: 
<img width="503" alt="image" src="https://github.com/woocommerce/woocommerce/assets/105309450/ac82b6f8-8ea2-4a24-9ea2-e325e6d4b239">

15. Once the translations have completed, you will see the following:
<img width="638" alt="image" src="https://github.com/woocommerce/woocommerce/assets/105309450/b9b3f7c5-b783-4611-b1fb-5bef0610ab9f">

16. You can now execute the e2e tests against the Espanol site by performing the following:
`export LANGUAGE=es_ES` to set the environment variable which will be used to run the tests against the Espanol site
17. `pnpm test:e2e-pw` to execute the e2e tests
18. Ensure all tests operate as expected
19. The other languages can be tested in turn. It is better to start with a fresh site so perform `pnpm env:restart` then repeat steps 4-18 to test in one of the following languages: 
- English (use `export LANGUAGE=en_GB` for step 16)
- French (use `export LANGUAGE=fr_FR` for step 16)
- Arabic (use `export LANGUAGE=ar_AR` for step 16)
